developer-workflow/extension-modules.rst

```diff
@@ -593,34 +593,14 @@ by executing :cpy-file:`Tools/build/regen-configure.sh`:
 ``make regen-configure`` and missing permissions
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Since this rule requires `Docker <https://docs.docker.com/desktop>`_ to be
-running, the following can be done on Linux platforms (``systemctl``-based):
```

Comment:
I think this section should:
1. clarify if `regen-configure` works with both `podman` and `docker`;
2. clarify which one takes the precedence;
3. recommend to (install and) use `podman`, since it should be simpler thank `docker`
4. link to the SO question in case the user is running into permission problems with `docker`

You might also want to update the section title to something more generic.

Reply:
This section was more for troubleshooting since those are the issues I encountered actually :') I can however say that `podman` takes precedence. This section was really for the specific issue of missing permissions (which was solved via the SO post). I think it's better if each issue has its own section together with its resolution (it helps when you encounter some issue). 

I'll put another section which recommends podman in general and explains that podman and docker are interchangeable modulo those weird docker issues.

Reply:
I've decided to take your approach, thereby simplifying the troubleshooting section (and make it only relevant for Docker-based configurations, not Podman ones).